src/action-ros-ci.ts

```diff
@@ -506,7 +512,7 @@ done`;
 	core.addPath(path.join(rosWorkspaceDir, "install", "bin"));
 
 	// Source any installed ROS distributions if they are present
-	let colconCommandPrefix = "";
+	let colconCommandPrefix = "cd src/ && ";
```

Comment:
As I felt the current implementation doesn't work with repositories that have only one package on top, move to `src/` as a test.

In detail, if the directory structure is like this, 

```txt
package_a/package.xml
package_a/ws/package_b/package.xml
```

`colcon list` can't find `package_b`.

```sh
$ cd package_a
$ colcon list
package_a	.	(ros.ament_cmake)
```

Reply:
How to reproduce.

```sh
mkdir -p ~/tmp
cd ~/tmp
ros2 pkg create package_a
mkdir -p package_a/ws
cd package_a/ws
ros2 pkg create package_a
ros2 pkg create package_b
colcon list
# package_a	package_a	(ros.ament_cmake)
# package_b	package_b	(ros.ament_cmake)
cd ../
colcon list
# package_a	.	(ros.ament_cmake)
```

Reply:
> ```
> package_a/package.xml
> package_a/ws/package_b/package.xml
> ```

I don't think that's a valid directory structure for packages. `package_b` should not be under `package_a`'s directory.

Reply:
Yes, but `action-ros-ci` doesn't do that? :thinking: (I don't understand the behavior perfectly yet.)

Reply:
Why/how are you getting a directory structure like this?

Reply:
Hmm... I'm trying to investigate what is happening... :thinking: 
https://github.com/kenji-miyake/action-ros-ci-test-condition/pull/19
https://github.com/kenji-miyake/action-ros-ci-test-condition/pull/20

Reply:
For some reason, the test results are different from what I expected.
So I suspect there would be this directory structure, but I'm not sure yet. I'm just testing.

Reply:
I'm not sure if it's related to your issue, but `action-ros-ci` creates a workspace and uses it as the current working directory for all commands. Tthis way, if the repo if checked out, it doesn't change anything. That's also why the current repo is cloned manually using a `.repos` file. https://github.com/ros-tooling/action-ros-ci/blob/bedd8c14fc9f0f8399667183e3633c431d9a4c05/src/action-ros-ci.ts#L356-L360

Reply:
The problem is this.
https://github.com/kenji-miyake/action-ros-ci-test-condition/pull/20#issuecomment-897040255

Reply:
> For some reason, the test results are different from what I expected.
> So I suspect there would be this directory structure, but I'm not sure yet. I'm just testing.

Can you point me to those test results?

Reply:
@christophebedard Could you show me an example of the directory structure that `action-ros-ci` creates?
Is it like this? :thinking: 

```txt
target_repo/.git
target_repo/src/target_repo (cloned by `package.repo`)
target_repo/src/depend_package1
target_repo/src/depend_package2
target_repo/src/...
```

Reply:
> The problem is this.
> [kenji-miyake/action-ros-ci-test-condition#20 (comment)](https://github.com/kenji-miyake/action-ros-ci-test-condition/pull/20#issuecomment-897040255)

You're using `condition="$ROS_VERSION == 2` for `example_interfaces`, but you're using `action-ros-ci@v0.2`, so it's not supposed to work, no?

Reply:
> it shouldn't know that example_interfaces is a dependency,

mm...? Since `source /opt/ros/galactic/setup.sh` is run and `example_interfaces` is under the workspace, I think `colcon` should detect the repository and `action-ros-ci` isn't related in this point.
(As I'm sleepy, will consider again tommorow. :sleeping: Thank you for your support!)

Reply:
For https://github.com/kenji-miyake/action-ros-ci-test-condition/pull/11, I don't think there is a `/opt/ros/galactic/setup.sh` file because `setup-ros` doesn't install any ROS 2 binaries: https://github.com/kenji-miyake/action-ros-ci-test-condition/blob/d3e2929016caa70fb2b718be5b861ce560805ee6/.github/workflows/build_and_test.yml#L31

Reply:
Same for https://github.com/kenji-miyake/action-ros-ci-test-condition/pull/20: https://github.com/kenji-miyake/action-ros-ci-test-condition/blob/67c3ef2d338bcb7122ae043e78bc519770e07ec0/.github/workflows/build_and_test.yml#L31

Reply:
Thank you for your information. I understand the behavior of `options: im.ExecOptions`, it turns out https://github.com/ros-tooling/action-ros-ci/pull/700#discussion_r687064211 was my misunderstanding because the cwd is set to `ros_ws`.

Reply:
I think there IS `/opt/ros/galactic/setup.sh` because it's printed here. (I assume it's printed only when the file exists.)
However, it doesn't set `ROS_VERSION`.
![image](https://user-images.githubusercontent.com/31987104/129396334-486c5846-5c14-4066-88f6-fc6e05d5430d.png)

Of course, if I use binary packages, it's set.
![image](https://user-images.githubusercontent.com/31987104/129396401-16f5f219-c969-4582-a41e-35a9e648a9b6.png)

So I'll debug the image's status using this action.
https://github.com/mxschmitt/action-tmate

Reply:
I think it's because the `rosdep` step installs some binary dependencies (like `ros-rolling-fastcdr` if you look at the output of `install_rosdeps.sh`) so there is indeed an `/opt/ros/$DISTRO/setup.sh` file, but `ros-$DISTRO-ros-environment` still isn't installed so the `ROS_*` environment variables aren't set (even if `/opt/ros/$DISTRO/setup.sh` exists and is sourced).

Reply:
I see, it's probably correct.
I'm considering how to solve this problem... :thinking: 
One simple solution would be to add prefix to these commands, but I feel it's not so good solution.
![image](https://user-images.githubusercontent.com/31987104/129397793-bb339dc1-e3db-4f3f-874d-3f89b56284b8.png)

@christophebedard Do you have any idea?

Reply:
Although I tried this one, since it's `bash -c`, it couldn't set envs.
![image](https://user-images.githubusercontent.com/31987104/129397941-61eb584a-3b59-4af7-917e-023f341881f2.png)

Reply:
I think you can set the `ROS_*` environment variables (no matter if it's a binary or source build) using the `options` object, like this: https://github.com/ros-tooling/action-ros-ci/blob/7ff80b89366718de2a315b5e9d9201edb81667a4/src/action-ros-ci.ts#L362-L365

Reply:
Ah, I understand. It's cool! I'll try.

Reply:
mm...it's not set... :pleading_face: 

Am I wrong?
https://github.com/kenji-miyake/action-ros-ci/blob/ae929b48837a24939db6672b744715a15af39749/src/action-ros-ci.ts#L367-L369

https://github.com/kenji-miyake/action-ros-ci/blob/ae929b48837a24939db6672b744715a15af39749/src/action-ros-ci.ts#L191
![image](https://user-images.githubusercontent.com/31987104/129402384-46f01eea-ccbf-4d1b-b60e-76a09e6fc388.png)